subsys/mgmt/smp_shell.c

```diff
@@ -96,18 +96,17 @@ bool smp_shell_rx_byte(struct smp_shell_data *data, uint8_t byte)
 	 * The received byte is part of an mcumgr command.  Process the byte
 	 * and return true to indicate that shell should ignore it.
 	 */
-	if (data->cur + data->end < sizeof(data->mcumgr_buff) - 1) {
```

Comment:
was this field never initialized?

Reply:
Not initialized explicitly. It was part of bss section, so was zero at the beginning. Then it was assigned 0 in other part of the code (after mcumgr_buff was filled with SMP fragment), but never to non-zero.